sos/report/plugins/openstack_mistral.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2022 Red Hat, Inc.
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin
+
+
+MISTRAL_DIRECTORIES = [
+        '/var/log/mistral/',
+        '/var/lib/mistral/',
+        ]
+MISTRAL_LOGS = [
+        '/var/log/mistral/*.log',
+        '/var/lib/mistral/*/*.log',
+        ]
+
+
+class OpenStackMistral(Plugin, RedHatPlugin):
+    '''Gather Mistral directories content, both data from /var/lib/mistral
+    and its log from /var/log/mistral if it exists (older OSP).
+    The data also embed logs for the ansible runs launched via the service,
+    meaning we'll be able to properly debug failures therein. The rest of the
+    data are the generated environment files, also really useful in order
+    to debug an issue at deploy or day-2 operations.
+    '''
+
+    short_desc = 'OpenStack Mistral'
+
+    plugin_name = "openstack_mistral"
+    profiles = ('openstack', 'openstack_undercloud')
+
+    def setup(self):
+        if self.get_option('all_log'):
+            self.add_copy_spec(MISTRAL_DIRECTORIES)
+        else:
+            self.add_copy_spec(MISTRAL_LOGS)
+
+
+class RedHatRDOManager(OpenStackMistral, RedHatPlugin):
```

Comment:
Ah, let's move that package thing there then. Would be better/more readable. Note: OSP *is* redhat specific - it's a product. The way we use mistral there is also specific to OSP. Hence the need to get files.

Lemme check some of the other plugins related to openstack, and I'll push something shortly.

Reply:
Could even do it in  a better way, by filtering using the "containers" keyword. Lemme know if it suits better.

Reply:
If this component *only* runs in a container, then yes that is a great way to control plugin enablement.

If it can also run outside of a container, then having the `packages = ('openstack-selinux',)` control _as well_ as the `containers` control would be best.